dietpi/boot

```diff
@@ -108,9 +108,6 @@
 	# Pre-installed image, 1st run
 	if (( $G_DIETPI_INSTALL_STAGE == 10 )); then
 
-		# - Set swap
-		/DietPi/dietpi/func/dietpi-set_swapfile $(grep -m1 '^[[:blank:]]*AUTO_SETUP_SWAPFILE_SIZE=' /DietPi/dietpi.txt | sed 's/^[^=]*=//') "$(grep -m1 '^[[:blank:]]*AUTO_SETUP_SWAPFILE_LOCATION=' /DietPi/dietpi.txt | sed 's/^[^=]*=//')"
```

Comment:
This needs to stay. First run setup does not run on pre-installed images. But the swapfile or course we do not want to ship with the image file/download, so it needs to be re-created.

But @Fourdee will know more about this, since I never created pre-installed images. Perhaps for those it still makes sense to run all first run setup steps, just have related software installed already, which should not be affected by first run setup steps. Only thing is that users might be confused about the dietpi-software prompt.

Reply:
Yep needs to stay as it was, for pre-made images. Currently only Allo images, but still needs to be there for all devices.

Reply:
oh! So far Lysmarine will setup much like Allo. I guess im gonna need it too... lol
I didin't tough about this use case. 

There is to 2 call of  `dietpi-set_swapfile` 
One on `dietpi/boot` and one inside the function `Apply_DietPi_FirstRun_Settings()`. 

**My elements of concern are :** 

I wish that one could be enough. 

I have a strong feeling that once compressed a pre-installed image don't take more space with a swap file in it. 

But we can't assume there will be enough space on the User drive and there is no way to predict this. 

It's over kill to check if we should create a swapfile at every boot.

My hypothesis is either it's not needed in `Apply_DietPi_FirstRun_Settings()` or it should be seperated in it's own one-time service. like `dietpi-fs_partition_resize.service` and `dietpi-firstboot.service`

Reply:
@FredericGuilbault 
Jep, `Apply_DietPi_FirstRun_Settings()` is not called when `(( $G_DIETPI_INSTALL_STAGE == 10 ))` (pre-installed image) and the other way round. So both need to create the default swap file.

> I have a strong feeling that once compressed a pre-installed image don't take more space with a swap file in it.

Not sure if the swap file is emptied on shutdown, but I guess not, instead it is cleared on boot when initialised most likely. But the main reason against pre-creating it:
- Some SBCs can have multiple RAM size options, e.g. RPi with 256/512/1024M versions and NanoPi M4 at least can have different RAM sizes as well.
- As we by default aim to reach min. 2G overall RAM, this then requires different or even no swap file.

> I wish that one could be enough.

As said, actually the firstrun setup steps should not hurt on pre-installed images as well. Even with Allo/Amiberry/Lysmarine files and services installed, it makes sense to apply locale, timezone, keyboard layout and especially do security relevant things like SSH host key regeneration and machine ID reset.
And network setup according to dietp.txt choices.
Actually I don't see any step that could/should affect a pre-installed software title.

@Fourdee 
Not sure in case of Allo, but perhaps we can or even should remove pre-installed images special handling? As said only the `dietpi-software` prompt on first login could confuse or be skipped when having the desired set pre-installed. Although it does not hurt to offer installing additional software on top.
But if wanted, we could simply solve this via `/DietPi/dietpi/.skip_firstrun_installs` flag file which makes DietPi-Login skip dietpi-software execution on first login. What you think?

Reply:
> As said, actually the firstrun setup steps should not hurt on pre-installed images as well. Even with Allo/Amiberry/Lysmarine files and services installed, it makes sense to apply locale, timezone, keyboard layout and especially do security relevant things like SSH host key regeneration and machine ID reset.
And network setup according to dietp.txt choices.
Actually I don't see any step that could/should affect a pre-installed software.

Yep, The fact that it can be re-run without breaking anythig is a proof of good design. 
I now understand for the security stuff the need to run it even if it been pre-installed.

> As said only the dietpi-software prompt on first login could confuse or be skipped when having the desired set pre-installed 

Yeah that's one of my big consern, A fast, question-less and reboot-less first boot is a major point.  My userbase is not techy at all. Their expectation is you plug it and you can move a mouse 20sec later. I had to sell pre-installed SDcard to people overwhelmed by RuFus and etcher...

> we could simply solve this via /DietPi/dietpi/.skip_firstrun_installs 

That kind of mecanism could effectively do the job, I mean it's the goal. But idk about creating an other dot file. Would it be possible to use DIETPI_INSTALL_STAGE   ?

Reply:
@FredericGuilbault 
> I had to sell pre-installed SDcard to people overwhelmed by RuFus and etcher...

Ui, in such cases I am wondering if running an on SBC is such a good idea 😉. However learning by doing, was the same for me when I started with an RPi and ownCloud that day.

> Would it be possible to use DIETPI_INSTALL_STAGE ?

Currently I don't know how this could work fine. DietPi-Preboot (now DietPi-Firstboot) set DIETPI_INSTALL_STAGE=0, the first update =1, then it's always the same when DietPi-Software is reached. E.g. using a different start value means a more complicated handling withing this scripts. Since currently the only difference is indeed the DietPi-Software prompt and as of discussion it makes sense to keep difference as small as possible, an additional flag file is the cleanest method. It can be simply removed after it did it's job by the script:
```
if [[ -f /DietPi/dietpi/.skip_firstrun_installs ]]; then
    rm -f /{DietPi,boot}/dietpi/.skip_firstrun_installs
else
    Promt DietPi-Software for first run installs
fi
```

Reply:
@FredericGuilbault @Fourdee 
I have a simple solution: `/DietPi/dietpi/.installed`
- This file is created on first `dietpi-software` run, so it actually serves as a flag file already.
- All current pre-installed images ship software that is selectable in `dietpi-software`, so this file needs to be pre-created.
- Since actually all other firstrun setup steps make sense for pre-installed images as well, including the ones currently done in first `dietpi-software` execution, we could simply adjust the login prompt a bid, to make it clearer to users what to do e.g.:
```
if [[ $G_DIETPI_INSTALL_STAGE == 1 ]]; then
    prompt_text='Please select software to install, or, select "Install" to finish with a minimal system.'
    [[ -f /DietPi/dietpi/.installed ]] && prompt_text="Your DietPi image has the following software titles pre-installed: <show list, excluding Dropbear and DietPi-RAMlog, which are always pre-installed>"
    G_WHIP_MSG "$prompt_text"
    ...
fi
```